bindings/pyroot/pythonizations/python/ROOT/_facade.py

```diff
@@ -9,13 +9,186 @@
 from cppyy import cppdef, include
 from libROOTPythonizations import gROOT
 from cppyy.gbl import gSystem
+import cppyy
 import cppyy.ll
 
 from ._application import PyROOTApplication
 from ._numbadeclare import _NumbaDeclareDecorator
 
 from ._pythonization import pythonization
 
+byteswap = ""
+
+
+def _CheckEndianessFromTypestr(typestr):
+    import ROOT
+
+    global byteswap
+
+    if len(byteswap) == 0:
+        ROOT.gInterpreter.Declare(
+            """
+std::string PyROOT_getByteswap()
+{
+#ifdef R__BYTESWAP
+    return "<";
+#else
+    return ">";
+#endif
+}
+"""
+        )
+        byteswap = ROOT.PyROOT_getByteswap()
+
+    endianness = typestr[:1]
+
+    if endianness != byteswap:
+        raise RuntimeError(
+            "Object not convertible: Endianess of __array_interface__['typestr'] does not match endianess of ROOT."
+        )
+
+
+def _GetCppTypeFromNumpyType(dtype):
+    if dtype == "i4":
+        return "int"
+    elif dtype == "u4":
+        return "unsigned int"
+    elif dtype == "i8":
+        return "Long64_t"
+    elif dtype == "u8":
+        return "ULong64_t"
+    elif dtype == "f4":
+        return "float"
+    elif dtype == "f8":
+        return "double"
+
+    raise RuntimeError("Object not convertible: Python object has unknown data-type '" + dtype + "'.")
```

Comment:
This is a good chance to convert this function to a Python dictionary which would be much more idiomatic